internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -1,185 +1,84 @@
-
-func CreateMigrateScript(proj projecter) error {
-	scriptValue := migrateNixScript
-	scriptLanguage := "bash"
-	switch runtime.GOOS {
-	case "darwin":
-		scriptValue = migrateMacScript
-	case "windows":
-		scriptValue = migrateWinScript
-		scriptLanguage = "powershell"
+// Migrate returns the given project's commit ID in either the new format (commit file), or the old
+// format (activestate.yaml).
+// If you require the commit file, use localcommit.Get().
+func (m *migrator) Migrate(pjpath string) (strfmt.UUID, error) {
+	logging.Debug("Migrating project to new localcommit format: %s", pjpath)
+	if err := m.setupProject(pjpath); err != nil {
+		return "", err
 	}
 
-	script := projectfile.Script{
-		projectfile.NameVal{
-			Name:  "migrate-to-buildscripts",
-			Value: string(scriptValue),
-		},
-		projectfile.ScriptFields{
-			Description: locale.T("projectmigration_script_description"),
-			Standalone:  true,
-			Language:    scriptLanguage,
-		},
-	}
+	configPath := filepath.Join(m.proj.Dir(), constants.ConfigFileName)
 
-	// We have to get a bit creative in writing the script, because calling `pjfile.Save()` will lead to reformatting
-	// of user curated yaml.
-	scriptB, err := yaml.Marshal(script)
+	// Add comment to activestate.yaml explaining migration
+	asB, err := fileutils.ReadFile(configPath)
 	if err != nil {
-		return errs.Wrap(err, "Could not marshal script")
+		return "", errs.Wrap(err, "Could not read activestate.yaml")
 	}
 
-	// Indent our script block
-	scriptB = bytes.Trim(scriptB, "\n")
-	lines := bytes.Split(scriptB, []byte("\n"))
-	for i, line := range lines {
-		prefix := "    "
-		if i == 0 {
-			prefix = "  - "
-		}
-		lines[i] = append([]byte(prefix), line...)
+	asB = append([]byte(locale.T("projectmigration_asyaml_comment")), asB...)
+	if err := fileutils.WriteFile(configPath, asB); err != nil {
+		return "", errs.Wrap(err, "Could not write to activestate.yaml")
 	}
-	scriptB = bytes.Join(lines, []byte("\n"))
-	scriptB = append(scriptB, []byte("\n")...)
 
-	// Splice it into the activestate.yaml
-	asB, err := fileutils.ReadFile(proj.Source().Path())
-	if err != nil {
-		return errs.Wrap(err, "Could not read activestate.yaml")
-	}
-
-	scriptsPos := scriptsRx.FindIndex(asB)
-	if scriptsPos != nil {
-		asB = append(asB[:scriptsPos[1]], append(scriptB, asB[scriptsPos[1]:]...)...)
-	} else {
-		asB = append(asB, append([]byte("\nscripts:\n"), scriptB...)...)
+	if !strfmt.IsUUID(m.proj.LegacyCommitID()) {
+		return "", locale.NewInputError("err_commit_id_invalid", m.proj.LegacyCommitID())
 	}
 
-	if err := fileutils.WriteFile(proj.Source().Path(), asB); err != nil {
-		return errs.Wrap(err, "Could not write activestate.yaml")
-	}
-
-	return nil
+	return strfmt.UUID(m.proj.LegacyCommitID()), nil
 }
 
-// Only show once per state tool invocation
-var warned = false
-
-func Warn(proj projecter) error {
-	if warned {
-		return nil
+func (m *migrator) Set(pjpath string, commitID string) error {
```

Comment:
`migrator.Set()` by name is not very clear. Maybe rename to `migrator.SetLegacyCommitID()`?

Reply:
I feel like this is redundant. The whole package is to migrate from a legacy version, don't need the methods to repeat that.

I called the Get() function Migrate() because it does more than just Get the value. But for Set that doesn't apply.